lib/Transforms/IPO/StructFieldCacheAnalysisImpl.h

```diff
@@ -120,6 +135,14 @@ class StructFieldAccessManager {
     return LookupBFI(*Func)->getBlockProfileCount(BB);
   }
 
+  /// Retrieve branch probability information of a branch
+  Optional<double> getBranchProbability(const BasicBlock* FromBB, const BasicBlock* ToBB) const{
+    assert(FromBB->getParent() == ToBB->getParent());
+    Function* func = const_cast<Function*>(FromBB->getParent());
+    auto Prob = LookupBPI(*func)->getEdgeProbability(FromBB, ToBB);
+    return 1.0 * Prob.getNumerator() / Prob.getDenominator();
```

Comment:
Use to change transform the number into double because the two getters only get integers.